File: frontend/src/services/task.service.ts
import api from './api';
import { Task } from '../types';

export const taskService = {
  getAllTasks: async (): Promise<Task[]> => {
    const response = await api.get<Task[]>('/tasks');
    return response.data;
  },

  getTaskById: async (id: number): Promise<Task> => {
    const response = await api.get<Task>(`/tasks/${id}`);
    return response.data;
  },

  createTask: async (task: Partial<Task>): Promise<Task> => {
    const response = await api.post<Task>('/tasks', task);
    return response.data;
  },

  updateTask: async (id: number, task: Partial<Task>): Promise<Task> => {
    const response = await api.put<Task>(`/tasks/${id}`, task);
    return response.data;
  },

  deleteTask: async (id: number): Promise<string> => {
    const response = await api.delete<string>(`/tasks/${id}`);
    return response.data;
  },

  // Admin only
  getAllTasksAdmin: async (): Promise<Task[]> => {
    const response = await api.get<Task[]>('/tasks/all');
    return response.data;
  },
  
  getTasksByUser: async (userId: number): Promise<Task[]> => {
    const response = await api.get<Task[]>(`/users/${userId}/tasks`);
    return response.data;
  }
};

File: backend/TaskManagement/src/main/java/com/taskmanagement/service/TaskService.java
package com.taskmanagement.service;

import com.taskmanagement.dto.TaskDto;
import java.util.List;

public interface TaskService {
    TaskDto createTask(TaskDto taskDto, Long userId);
    List<TaskDto> getAllTasksByUserId(Long userId);
    TaskDto getTaskById(Long taskId, Long userId);
    TaskDto updateTask(Long taskId, TaskDto taskDto, Long userId);
    void deleteTask(Long taskId, Long userId);
    List<TaskDto> getAllTasks();
}
